week2/homework/src/index.js

```diff
@@ -1,3 +1,122 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+let fs = require('fs');
+let command = process.argv[2];
+
+if (command === 'add') {
+    let item = process.argv[3];
+    add(item);
+} else if (command === 'delete') {
+    let remove = process.argv[3];
+    deleteItem(remove);
+} else if (command === 'list') {
+    process.argv[3];
+    list();
+} else if (command === 'reset') {
+    resetAll();
+} else if (command === 'update') {
+    let index = process.argv[3];
+    let newLine = process.argv[4];
+    update(index, newLine);
+} else if (command === 'help') {
+    helpSection();
+}
+
+function helpSection() {
+    fs.readFile('./help.txt', 'utf8', function(error, data) {
+        if (error) {
+            console.error(error);
+        }
+        console.error(data);
+    });
+}
+
+function update(index, newLine) {
+    fs.readFile('./data.txt', 'utf8', function(error, data) {
+        if (error) {
+            console.error(error);
+        }
+        let arrayData = data.split('\n');
+
+        console.log(arrayData[index] + ' updated =>: ' + newLine);
+
+
+        let split = [];
+        arrayData.forEach((item) => {
+            split.push(item + '\n');
+
+            let replaceItem = split.toString().replace(arrayData[index - 1], newLine);
+
+            fs.writeFile('./data.txt', replaceItem.replace(/,/g, ''), function(error) {
+                if (error) {
+                    console.log(error);
+                }
+
```

Comment:
This was a difficult part, but you came up with a solution that is good! This is the kind of tasks you will be doing a lot as a programmer, so if it is frustrating you, please let me know which parts are difficult or cumbersome and maybe I can explain them further for you. Don't think a library (like express) is fixing these kind of problems for you: these are the basics and it is good to really onderstand them well!